docs/builds/guides/frameworks/angular.md

```diff
@@ -299,6 +331,171 @@ Then, add it in the `angular.json` configuration file.
 }
 ```
 
+## Accessing the CKEditor API from an Angular Component
+
+The **ckeditor** component will provide all the functionality needed for most use cases.  In 
+cases where access to the Javascript-level {@link api CKEditor API} is needed
+it is easy to access with an additional step.
+
+To do this, assign the Angular **ckeditor** component to a locally scoped variable as
+follows:
+
+```html
+	<ckeditor #editor1 [editor]="Editor" [config]="ckconfig"
+			  [(ngModel)]="text" name="editcontent"></ckeditor>
+```
+
+You can then refer to that variable as needed in any method call within the component
+such as:
+
+```html
+	<button type="button" (click)="doPasteSomething(editor1)">Do Someting</button>
+```
+
+In the component the target function will look like this:
+
+```ts
+  doPasteSomething(editorComponent: CKEditorComponent) {
+    const editor = editorComponent.editorInstance;
+    editor.model.insertContent(' >> This just got pasted! << ');
+  }
+```
+
+The operations performed may be anything that is defined and allowable by the API.
+
+Note that the Angular **editor1** template variable may also be accessed with an
+**@ViewChild**-decorated variable declaration.  The method in the example, however,
+will be preferable in the case where there is more than one **ckeditor** element on
+the page.  Also, if the **ckeditor** element is inside an ***ngIf** structure the **@ViewChild**
+declared variable may be inadvertently undefined.
+
+
+## Implementing an Upload Adapter in an Angular Application
+
+An Upload Adapter can be used with CKEditor so that when a user adds an image 
+to a document it is encoded as a link to the image.  The image itself is stored
+separately from the document.   When images are large or numerous this results
+in noticable improvement in performance over the default behavior.  Read about custom
+UploadAdapters {@link framework/guides/deep-dive/upload-adapter here}. 
+
+The first step is to create an Upload Adapter.  This can be done in the same file where the 
+component that will use it.  Alternatively this can have its own file and be
+imported.
+
+This class will be instantiated each time that CKEditor needs it.  Note that
+the functions in this class will run outside an **NgZone** which means that
+if it makes changes to variables bound to HTML elements, the shadow-DOM logic
+in the Angular framework won't detect those changes.  
+
+The instance property **loader** is the primary interface between the UploadAdapter
+and the CKEditor instance.
+
+```ts
```

Comment:
I'd drop this part since the upload adapter can be written in different ways depending on the needs. It's OK to describe it and link to official docs. But the code should not be a part of this guide. There are no Angular-specific things here.